### ISSUE TITLE
Recent new type of Luck.Main.parse breaks the template examples

### ISSUE THREAD

**Issue by pirapira** [QuickChick/Luck#4]
A recent change 84d8bd30901b81ba57a3ec0696cd17e5b7d83e68 seems to have broken the build of `combinations.hs` example in `examples-tempalte` directory.

Before the change:
```
[12:41] yh@sri: ~/infra-src/Luck/luck(:c64d06e|…)$ stack build && stack runghc examples-template/combination.hs
There were multiple candidates for the Cabal entry "Outer.Parser" (/home/yh/infra-src/Luck/luck/src/Outer/Parser.y), picking /home/yh/infra-src/Luck/luck/src/Outer/Parser.hs
Just [1,0,0,0,0,0,0,0,0,0,1,1,0]
Just [1,0,0,0,0,0,0,0,1,1,0,0,0]
Just [0,0,1,0,0,1,0,0,0,1,0,0,0]
Just [0,0,0,0,1,0,1,1,0,0,0,0,0]
Just [1,0,0,0,0,0,0,0,0,0,1,1,0]
Just [0,0,1,1,0,0,0,0,0,0,0,0,1]
Just [0,0,0,0,0,1,0,1,1,0,0,0,0]
Just [0,0,0,0,0,1,1,0,0,0,0,1,0]
Just [0,0,1,0,0,0,0,0,0,0,0,1,1]
Just [0,1,0,0,1,0,0,0,0,1,0,0,0]
Just [1,0,0,0,0,0,1,0,0,1,0,0,0]
```

After the change:
```
[12:44] yh@sri: ~/infra-src/Luck/luck(master|…)$ git show | head -n 1
commit 84d8bd30901b81ba57a3ec0696cd17e5b7d83e68
[12:45] yh@sri: ~/infra-src/Luck/luck(master|…)$ stack build && stack runghc examples-template/combination.hs
There were multiple candidates for the Cabal entry "Outer.Parser" (/home/yh/infra-src/Luck/luck/src/Outer/Parser.y), picking /home/yh/infra-src/Luck/luck/src/Outer/Parser.hs

examples-template/combination.hs:6:9: error:
    • Couldn't match type ‘Data.ByteString.Internal.ByteString’
                     with ‘[Outer.AST.Decl]’
      Expected type: Outer.AST.Prg
        Actual type: Data.ByteString.Internal.ByteString
    • In the second argument of ‘Luck.Main.parse’, namely
        ‘Data.ByteString.Char8.pack
```

I used GHC 8.0.1 for both experiments.
```
$ stack ghc -- --version
The Glorious Glasgow Haskell Compilation System, version 8.0.1
```

**Comment by Lysxia** [comment]
Thanks for reporting this. This is now fixed upstream.

**Comment by pirapira** [comment]
Thank you very much.  It's my fault I didn't report this in the upstream in the first place.

**Comment by Lysxia** [comment]
It's fine! I don't think we really have a designated place for issues, so this mirror might be as good as any other.

**Comment by Lysxia** [comment]
Fixed since 2cb4727.

**Comment by pirapira** [comment]
Thank you!
